index.js

```diff
@@ -37,41 +38,163 @@ var EchoClient = function(config){
+ * @param  {object}   queryParams       Hash of parameters to add to the query
+ * @param  {boolean}  useCache          Indicates if cache should be used or not
+ * @callback callback
+ */
+EchoClient.prototype.queryAnalytics = function(token, queryOperator, queryParams, useCache, callback) {
+    if (!token) {
         throw new Error('Missing Persona token');
     }
-    if(!data){
-        throw new Error('Missing data');
+
+    if (!queryOperator) {
+        throw new Error('Missing Analytics queryOperator');
+    }
+
+    var validOperators = ['hits', 'average', 'sum', 'max', 'funnel'];
+
+    if (validOperators.indexOf(queryOperator) === -1) {
+        throw new Error('Invalid Analytics queryOperator');
     }
-    if(!data.class){
-        throw new Error('Missing field data.class');
+
+    if (!queryParams) {
+        throw new Error('Missing Analytics queryParams');
     }
-    if(!data.source){
-        throw new Error('Missing field data.source');
+
+    var constructQueryStringResponse = this._queryStringParams(queryParams);
+
+    if (constructQueryStringResponse.errors) {
+        throw new Error('Invalid Analytics queryParams');
     }
 
     var requestOptions = {
-        url: this.config.echo_endpoint+'/1/events',
+        url: this.config.echo_endpoint + '/1/analytics/' + queryOperator + '?' + constructQueryStringResponse.queryString,
+        method: 'GET',
         headers: {
-            'Accept': 'application/json',
-            'Authorization':'Bearer '+token
-        },
-        body:data,
-        method:'POST',
-        json:true
+            'Content-Type': 'application/json',
+            'Authorization': 'Bearer ' + token
+        }
     };
 
-    request.post(requestOptions, function(err, response, body){
-        if(err){
+    if (useCache === false) {
+        requestOptions.headers['cache-control'] = 'none';
+    }
+
+    this.debug(JSON.stringify(requestOptions));
+
+    request.get(requestOptions, function(err, response, body) {
+        if (err) {
             callback(err);
-        } else{
+        } else {
             callback(null, body);
         }
     });
+};
 
-    this.debug(JSON.stringify(requestOptions));
+/**
+ * Build up a query string
+ * @param {object} params
+ * @returns {string}
+ * @private
+ */
+EchoClient.prototype._queryStringParams = function(params) {
+    var queryString = '';
+    var queryStringParams = [];
+    var paramErrors = null;
+
+    var isValidParameter = function (parameter) {
+        var validParamters = [
+            'class',
+            'source',
+            'property',
+            'interval',
+            'group_by',
+            'key',
+            'value',
+            'from',
+            'to',
+            'percentile',
+            'user',
+            'filter',
+            'n'
+        ];
 
+        var hasDot = parameter.indexOf('.');
+
+        if (hasDot > -1) {
+            parameter = parameter.substring(0, hasDot);
+        }
```

Comment:
I'm not entirely sure what's going on here. None of the valid parameters above has dots in them?